git/git.go

```diff
@@ -39,6 +39,7 @@ const (
 	// A ref which can be used as a placeholder for before the first commit
 	// Equivalent to git mktree < /dev/null, useful for diffing before first commit
 	RefBeforeFirstCommit = "4b825dc642cb6eb9a060e54bf8d69288fbee4904"
+	ObjectIDRegex        = "(?:[0-9a-f]{40}(?:[0-9a-f]{24})?)"
```

Comment:
This is completely possible, as you've pointed out.  My goal with the capturing group is to future-proof things a little bit so that if we get a new caller we don't shoot ourselves in the foot by matching more or less than we expect.  So while we _can_, I think right here I'd prefer not to unless we discover a compelling performance justification.

Reply:
OK, sure -- I confess I don't quite see how the outer non-capturing group would help in that way (if it's interpolated into a capturing context, it's equivalent to having it not exist, and equally so if it's interpolated into a non-capturing context), but it can't hurt, obviously!

Reply:
Here's an example where something could go wrong:

```
Pattern = "abc"
FancyPattern = fmt.Sprintf("%s+", Pattern)
```

This regex is `abc+`, so it matches `abccccc` but not `abcabc`.  If we had put a non-capturing group around `Pattern`, it would match the latter.  This of course is solved if you capture in `FancyPattern`, and this case is less likely to occur in the Git context because we don't usually smash object IDs up next to each other and so would only be likely to match with `?`.

However, as someone who used to work extensively in Perl, I've been bitten by this numerous times and so prefer to be a little more cautious with regexes in this regard.

Reply:
Ah, I wasn't considering a case where you followed it up with a quantifier, with the goal of, perhaps, trying to match multiple SHAs in a row or something.   Thank you for the elucidation!